packages/prices/utils/price.ts

```diff
@@ -76,7 +76,7 @@ export const getCurrencyFromPriceResponse = (
 		| Record< string, never >
 		| CartShippingPackageShippingRate
 ): Currency => {
-	if ( ! currencyData?.currency_code ) {
```

Comment:
This prevents invalid (default state) `currencyData` taking precedence over the default `siteCurrencySettings`.